src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer.cs

```diff
@@ -305,18 +360,62 @@ bool IsCreateHelper(IMethodSymbol method)
+
+                if (titleArgument.Value.ConstantValue.HasValue &&
+                    titleArgument.Value.ConstantValue.Value is string title &&
+                    (title.Contains(".") || ContainsLineReturn(title)))
+                {
+                    operationAnalysisContext.ReportDiagnostic(titleArgument.CreateDiagnostic(DefineDiagnosticTitleCorrectlyRule));
+                }
+            }
+        }
+
+        private static void AnalyzeMessage(OperationAnalysisContext operationAnalysisContext, ImmutableArray<IArgumentOperation> creationArguments)
+        {
+            IArgumentOperation messageArgument = creationArguments.FirstOrDefault(a => a.Parameter.Name.Equals("messageFormat", StringComparison.OrdinalIgnoreCase));
+
+            if (messageArgument != null &&
+                messageArgument.Value.ConstantValue.HasValue &&
+                messageArgument.Value.ConstantValue.Value is string message)
+            {
+                var splitPeriod = message.Split('.');
+                // Report if the message contains a line return or ends with '.' and is not a multi-sentences message
+                if ((splitPeriod.Length == 2 && splitPeriod[splitPeriod.Length - 1].Length == 0) ||
+                    ContainsLineReturn(message))
+                {
+                    operationAnalysisContext.ReportDiagnostic(messageArgument.CreateDiagnostic(DefineDiagnosticMessageCorrectlyRule));
+                }
+            }
+        }
+
+        private static void AnalyzeDescription(OperationAnalysisContext operationAnalysisContext, ImmutableArray<IArgumentOperation> creationArguments)
         {
-            IParameterSymbol title = descriptorCreationMethod.Parameters.FirstOrDefault(p => p.Name == "title");
-            if (title != null &&
-                title.Type != null &&
-                title.Type.SpecialType == SpecialType.System_String)
+            IArgumentOperation descriptionArgument = creationArguments.FirstOrDefault(a => a.Parameter.Name.Equals("description", StringComparison.OrdinalIgnoreCase));
+
+            if (descriptionArgument != null &&
+                descriptionArgument.Value.ConstantValue.HasValue &&
+                descriptionArgument.Value.ConstantValue.Value is string description)
             {
-                Diagnostic diagnostic = creation.Value.CreateDiagnostic(UseLocalizableStringsInDescriptorRule, WellKnownTypeNames.MicrosoftCodeAnalysisLocalizableString);
-                reportDiagnostic(diagnostic);
+                var lastChar = description[description.Length - 1];
+                if (!lastChar.Equals('.') && !lastChar.Equals('!') && !lastChar.Equals('?'))
+                {
+                    operationAnalysisContext.ReportDiagnostic(descriptionArgument.CreateDiagnostic(DefineDiagnosticDescriptionCorrectlyRule));
+                }
             }
         }
 
+        private static bool ContainsLineReturn(string s)
+            => s.Contains("\r") || s.Contains("\n");
```

Comment:
I am not using `Environment.NewLine` because nothing guarantees user is respecting the OS newline convention.